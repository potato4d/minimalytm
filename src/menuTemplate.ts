import { App, MenuItemConstructorOptions } from 'electron'

export function createMenuTemplate(app: App): MenuItemConstructorOptions[] {
  return [
    {
      label: 'Minimal YouTube Music Player',
      submenu: [
        {
          label: 'About this application',
          selector: 'orderFrontStandardAboutPanel:'
        } as any,
        { type: 'separator' },
        {
          label: 'Quit',
          accelerator: 'Command+Q',
          click: () => {
            app.quit()
          }
        }
      ]
    },
    {
      label: 'Edit',
      submenu: [
        { label: 'Undo', accelerator: 'CmdOrCtrl+Z', selector: 'undo:' },
        { label: 'Redo', accelerator: 'Shift+CmdOrCtrl+Z', selector: 'redo:' },
        { label: 'Cut', accelerator: 'CmdOrCtrl+X', selector: 'cut:' },
        { label: 'Copy', accelerator: 'CmdOrCtrl+C', selector: 'copy:' },
        { label: 'Paste', accelerator: 'CmdOrCtrl+V', selector: 'paste:' },
        { type: 'separator' },
        {
          label: 'Select All',
          accelerator: 'CmdOrCtrl+A',
          selector: 'selectAll:'
        }
      ]
    }
  ]
}
