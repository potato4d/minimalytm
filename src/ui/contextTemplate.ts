import { App, MenuItemConstructorOptions } from 'electron'

export function createContextTemplate(app: App): MenuItemConstructorOptions[] {
  return [
    {
      label: 'About this application',
      selector: 'orderFrontStandardAboutPanel:'
    } as any,
    {
      label: 'Quit',
      accelerator: 'Command+Q',
      click: () => {
        app.quit()
      }
    }
  ]
}
