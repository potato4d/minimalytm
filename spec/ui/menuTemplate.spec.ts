import { createMenuTemplate } from '../../src/ui/menuTemplate'

describe('menuTemplate', () => {
  test('createMenuTemplate()', () => {
    expect(createMenuTemplate({} as any)).toMatchSnapshot()
  })
})
