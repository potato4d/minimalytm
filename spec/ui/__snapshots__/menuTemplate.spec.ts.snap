// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`menuTemplate createMenuTemplate() 1`] = `
Array [
  Object {
    "label": "Minimal YouTube Music Player",
    "submenu": Array [
      Object {
        "label": "About this application",
        "selector": "orderFrontStandardAboutPanel:",
      },
      Object {
        "type": "separator",
      },
      Object {
        "accelerator": "Command+Q",
        "click": [Function],
        "label": "Quit",
      },
    ],
  },
  Object {
    "label": "Edit",
    "submenu": Array [
      Object {
        "accelerator": "CmdOrCtrl+Z",
        "label": "Undo",
        "selector": "undo:",
      },
      Object {
        "accelerator": "Shift+CmdOrCtrl+Z",
        "label": "Redo",
        "selector": "redo:",
      },
      Object {
        "accelerator": "CmdOrCtrl+X",
        "label": "Cut",
        "selector": "cut:",
      },
      Object {
        "accelerator": "CmdOrCtrl+C",
        "label": "Copy",
        "selector": "copy:",
      },
      Object {
        "accelerator": "CmdOrCtrl+V",
        "label": "Paste",
        "selector": "paste:",
      },
      Object {
        "accelerator": "CmdOrCtrl+W",
        "label": "Close Window",
        "selector": "close:",
      },
      Object {
        "type": "separator",
      },
      Object {
        "accelerator": "CmdOrCtrl+A",
        "label": "Select All",
        "selector": "selectAll:",
      },
    ],
  },
]
`;
